src/main/java/com/google/ehub/servlets/EntertainmentItemServlet.java

```diff
@@ -0,0 +1,176 @@
+    response.getWriter().println(new Gson().toJson(entertainmentItemList));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String title = request.getParameter(TITLE_PROPERTY_KEY);
+    String description = request.getParameter(DESCRIPTION_PROPERTY_KEY);
+    Optional<String> imageURLOptional =
+        BlobstoreURLUtility.getUploadURL(request, IMAGE_URL_PROPERTY_KEY);
+
+    if (!arePOSTRequestParametersValid(title, description, imageURLOptional)) {
+      return;
+    }
+
+    Entity entertainmentItemEntity = new Entity(ENTERTAINMENT_ITEM_KIND);
+    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, title);
+    entertainmentItemEntity.setProperty(LOWERCASE_TITLE_PROPERTY_KEY, title.toLowerCase());
+    entertainmentItemEntity.setProperty(DESCRIPTION_PROPERTY_KEY, description);
+    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, imageURLOptional.get());
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(entertainmentItemEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Creates a query to search for a list of entertainment items with appropriate
+   * search filters and sorting order.
+   *
+   * @param searchValue the search value used by the query
+   * @param sortingMode the sorting mode used by the query
+   * @return query built from the respective filter and sorting parameters
+   */
+  Query createEntertainmentItemQuery(String searchValue, String sortingMode) {
+    Query query = new Query(ENTERTAINMENT_ITEM_KIND);
+
+    if (!searchValue.isEmpty()) {
+      FilterPredicate greaterThanOrEqualFilter = new FilterPredicate(LOWERCASE_TITLE_PROPERTY_KEY,
+          FilterOperator.GREATER_THAN_OR_EQUAL, searchValue.toLowerCase());
+
+      FilterPredicate lessThanFilter =
+          new FilterPredicate(LOWERCASE_TITLE_PROPERTY_KEY, FilterOperator.LESS_THAN,
+              NextIncreasingLexicographicStringUtility.getNextIncreasingLexicographicString(
+                  searchValue.toLowerCase()));
+
+      query =
+          query.setFilter(CompositeFilterOperator.and(greaterThanOrEqualFilter, lessThanFilter));
+    }
+
+    if (sortingMode.equals(TITLE_INCREASING_ALPHA_KEY)) {
+      query = query.addSort(LOWERCASE_TITLE_PROPERTY_KEY, SortDirection.ASCENDING);
+    } else if (sortingMode.equals(TITLE_DECREASING_ALPHA_KEY)) {
+      query = query.addSort(LOWERCASE_TITLE_PROPERTY_KEY, SortDirection.DESCENDING);
+    }
+
+    return query;
+  }
+
+  /**
+   * Verifies if the parameters given in the HTTP GET Request are valid.
+   *
+   * @param searchValue the search value given in the GET request parameter
+   * @param sortingMode the sorting mode given in the GET request parameter
+   * @return true if the parameters given in the GET request are valid, false otherwise
+   */
+  private static boolean areGETRequestParametersValid(String searchValue, String sortingMode) {
```

Comment:
Nit: style guide for acronyms: https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

`areGetRequestParametersValid`
`arePostRequestParametersValid`
etc

Reply:
I removed capitalizations in last commit.